extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -698,6 +780,137 @@ angular.module('oppia').component('svgFilenameEditor', {
+            angle: 0
+          };
+          ctrl.pieChartDataInput.push(dataInput);
+          $scope.$applyAsync();
+        }
+      };
+
+      var getPieSlice = function(center, radius, startAngle, endAngle, color) {
+        var angle = endAngle - startAngle;
+        var halfAngle = angle / 2;
+        var halfChord = radius * Math.sin(angle / 2);
+        var height = Math.sqrt(Math.pow(radius, 2) - Math.pow(halfChord, 2));
+        var radiansToDegrees = 180 / Math.PI;
+
+        var arc = new fabric.Circle({
+          radius: radius,
+          startAngle: -halfAngle,
+          endAngle: halfAngle,
+          left: center.x,
+          top: center.y,
+          originX: 'center',
+          originY: 'center',
+          fill: color,
+          stroke: color,
+          strokeWidth: 1,
+          strokeUniform: true,
+          id: 'group' + ctrl.groupCount
+        });
+        arc.toSVG = ctrl.customToSVG(arc.toSVG, 'path', arc.id);
+        var p1 = new polyPoint (height + center.x, center.y + halfChord);
+        var p2 = new polyPoint (height + center.x, center.y - halfChord);
+        var tri = new fabric.Polygon([center, p1, p2, center], {
+          fill: color,
+          stroke: color,
+          strokeWidth: 1,
+          strokeUniform: true,
+          id: 'group' + ctrl.groupCount
+        });
+        tri.toSVG = ctrl.customToSVG(tri.toSVG, tri.type, tri.id);
+        var rotationAngle = (startAngle + halfAngle) * radiansToDegrees;
+        var slice = new fabric.Group([arc, tri], {
+          originX: 'center',
+          originY: 'center',
+          top: center.y,
+          left: center.x,
+          angle: rotationAngle,
+        });
+        return slice;
+      };
+
+      var getTextIndex = function(text, lineNum, charIndex) {
+        return (
+          text.split('\n').slice(0, lineNum).reduce(function(sum, textLine) {
+            return sum + textLine.length + 1;
+          }, 0) + charIndex);
+      };
+
+      var createChart = function() {
+        var total = 0;
+        var currentAngle = 0;
+        var pieSlices = [];
+        var legendText = '';
+        for (var i = 0; i < ctrl.pieChartDataInput.length; i++) {
+          total += ctrl.pieChartDataInput[i].data;
+          legendText += '\u2587 - ';
+          legendText += (
+            ctrl.pieChartDataInput[i].name + ' - ' +
+            ctrl.pieChartDataInput[i].data + '\n');
+        }
+        legendText = legendText.slice(0, -1);
+        for (var i = 0; i < ctrl.pieChartDataInput.length; i++) {
+          ctrl.pieChartDataInput[i].angle = (
+            ctrl.pieChartDataInput[i].data / total * Math.PI * 2);
+          pieSlices.push(getPieSlice(
+            new polyPoint(50, 50), 30, currentAngle,
+            currentAngle + ctrl.pieChartDataInput[i].angle,
+            ctrl.pieChartDataInput[i].color));
+          currentAngle += ctrl.pieChartDataInput[i].angle;
+        }
+        ctrl.fabricjsOptions.size = '18px';
```

Comment:
Why are we doing this? Will this affect other tools too?

Reply:
This is to prevent the text from being too small. This can be changed in the editor using the size dropdown. This doesnot affect other shapes.

Reply:
Can you please add a comment in the code so that it doesn't get removed accidentally in the future?

Reply:
I have already added the comment.

Reply:
Ah I see. Thanks!